skimage/graph/_mcp.pyx

```diff
@@ -391,9 +391,13 @@ cdef class MCP:
         """
         pass
 
-
+    @deprecate_parameter(
+        "max_cumulative_cost",
+        start_version="0.25",
+        stop_version="0.27",
+    )
     def find_costs(self, starts, ends=None, find_all_ends=True,
-                   max_coverage=1.0, max_cumulative_cost=None, max_cost=None):
+                   max_coverage=1.0, max_cumulative_cost=DEPRECATED, max_step_cost=None):
```

Comment:
Almost forgot: good opportunity to make `max_step_cost` keyword only with inserting `*,` before it. We've been slowly migrating towards that kind of API so that each function has fewer positional keywords.

Reply:
Changed it so that starting from `find_all_ends` only keyword arguments are allowed.

Reply:
Oh, I probably didn't explain that properly. We can only make the new parameter keyword-only. If we turn `find_all_ends` and following into keyword-only ones, we potentially break existing users without warning. :sweat_smile: 

Fixed in https://github.com/scikit-image/scikit-image/pull/7625/commits/815caec25f32265491e8f3656316737e75b7de07.